### PR TITLE
EVM-C renames

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: include/evm.h [hunk 1]
```diff
@@ -328,25 +328,42 @@ EXPORT struct evm_result evm_execute(struct evm_instance* instance,
...
-EXPORT void evm_destroy_result(struct evm_result);
+/// Releases the resources assigned to the execution result.
+///
+/// This function releases memory (and other resources, if any) assigned to the
+/// specified execution result making the result object invalid.
+///
+/// @param result  The execution result which resource are to be released. The
+///                result itself it not modified by this function, but becomes
+///                invalid and user should discard it as well.
+EXPORT void evm_release_result_resources(struct evm_result const* result);
```
Comment (axic) [inline include/evm.h]: Maybe just call it `evm_release_result`?

**Comment by axic** [comment]
Based on the implementation my takeaway is:
- `evm_prepare_code` can be optionally called to create a compiled version of the code identified by the `code_hash` as a key.
- `evm_get_code_status` tells the status associated with a `code_hash`
- `evm_execute` will use the cache if there's one for the given `code_hash`

Is that correct? Can you please document it in evm.h?

**Comment by chfast** [comment]
Yes, this is correct.

File: include/evm.h (hunk 1 above)
Comment (axic) [inline include/evm.h]: So who frees `evm_result` then? It is allocated by the callee so it should be freed by the callee (practically as part of `evm_release_result`).

**Comment by chfast** [comment]
I will extend the documentation in this area after adding vtables.

File: include/evm.h (hunk 1 above)
Comment (chfast) [inline include/evm.h]: `evm_result` is owned by the caller of `evm_execute()`.

**Pull request merged**